include/swift/AST/DiagnosticEngine.h

```diff
@@ -734,6 +758,11 @@ namespace swift {
       return diagnosticDocumentationPath;
     }
 
+    void setLocalization(std::string locale, std::string path) {
+      if (!locale.empty() && !path.empty())
```

Comment:
Should this `assert` instead of silent just not adding the path and locale values? This is just a question :)

Reply:
I don’t quite understand you. But the idea here that we might run the compiler without the frontend flags so if that’s the case we don’t need to initialize `localization`. Do you have a better thing in mind other than checking in `locale` and `path` are empty?

Reply:
Ah, I was just thinking about the call to the function ... because the way it is it right now it just don't add the locale if one of the values are not valid, so my comment was more about fail with assert or warn in some way so the caller knows that it is trying to pass invalid values, instead of just silent not adding and let the caller wondering why it is not adding... 
But it was more a question anyways :)

Reply:
@xedin Does this make sense in this case?

Reply:
@LucianoPAlmeida I think this could be an `assert` sure.
 @HassanElDesouky would add sanitization for locale/path in his next PR, it's going to be done before this method is called, this logic could be adjusted then.

Reply:
That's great, thanks @xedin :)